pkg/controller/master/vmimagedownloader/vmimagedownloader_controller.go

```diff
@@ -281,6 +284,61 @@ func (h *vmImageDownloaderHandler) getClusterRepoImage() string {
 	return ""
 }
 
+func (h *vmImageDownloaderHandler) ReconcileDeploymentOwners(_ string, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if deployment, ok := obj.(*appsv1.Deployment); ok {
+		for _, ownerReference := range deployment.GetOwnerReferences() {
+			if ownerReference.Kind == kindVirtualMachineImageDownloader {
+				return []relatedresource.Key{
+					{
+						Namespace: deployment.Namespace,
+						Name:      ownerReference.Name,
+					},
+				}, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func (h *vmImageDownloaderHandler) genInitContainer(mode *corev1.PersistentVolumeMode, vmImageName string) corev1.Container {
+	virtImageStr := h.getVirtHandlerImage()
+	initContainer := corev1.Container{
+		Name:            "image-coverter",
+		Image:           virtImageStr,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command:         []string{"/bin/sh", "-c"},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "image-dir",
+				MountPath: "/image-dir",
+			},
+		},
+	}
+
+	targetImgVolPath := "/tmp/image-vol"
+	convertSrcPath := targetImgVolPath
+	if *mode == corev1.PersistentVolumeFilesystem {
+		convertSrcPath = fmt.Sprintf("%s/disk.img", targetImgVolPath)
+	}
+	if *mode == corev1.PersistentVolumeBlock {
+		initContainer.VolumeDevices = []corev1.VolumeDevice{
+			{
+				Name:       "image-vol",
+				DevicePath: targetImgVolPath,
+			},
+		}
+	} else if *mode == corev1.PersistentVolumeFilesystem {
+		initContainer.VolumeMounts = append(initContainer.VolumeMounts, corev1.VolumeMount{
+			Name:      "image-vol",
+			MountPath: targetImgVolPath,
+		})
+	}
+	convertCmd := fmt.Sprintf("qemu-img convert -t none -T none -W -m 8 -f raw %s -O qcow2 -c -S 4K /image-dir/%s.qcow2", convertSrcPath, vmImageName)
```

Comment:
any command to check if the image has already met format, instead of alway run convert ?

Reply:
We could ensure the source is in the raw format because this is created ourselves.
The source is the 3rd-party golden image pvc. This is the raw format.